packages/zk-transfer-web/src/App.tsx

```diff
@@ -7,18 +7,35 @@ import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
 import Homepage from './views/Home';
 import MakeTransaction from './views/MakeTransaction';
 import Democracy from './components/context/Democracy';
+import { getScreenName, setScreenName, generateScreenName } from "./util";
 
 type AppProp = {
   demo: any
 };
 
 function App({ demo } : AppProp) {
-  const [chainId, setDemoChain] = useState(demo.chainId);
+  const [state, setState] = useState({
+    chainId: demo.chainId,
+    screenName: ''
+  });
 
   useEffect(() => {
+    const { chainId } = state;
+
     if (!chainId) {
       demo.clientInit()
-        .then(() => setDemoChain(demo.chainId));
+        .then(() => {
+          const screenName = getScreenName(demo.chainId);
+
+          if (!screenName) {
```

Comment:
We could use the same screenname across all chain IDs, but it works fine this way for now.